hail/python/hail/expr/types.py

```diff
@@ -1019,6 +1021,83 @@ def clear(self):
         for f, t in self.items():
             t.clear()
 
+class tunion(HailType, Mapping):
+    @typecheck_method(case_types=hail_type)
```

Comment:
Can you add some documentation? I'm not sure what a union type is. Is it the same thing as a C union type?